### DISCUSSION TITLE
How does the exclude option work

### DISCUSSION THREAD

**Discussion by OlaoluwaM** [o2sh/onefetch#1215] [Q&A]
What format does it accept? Strings? Regexes? Is there a way to differentiate between directories and files?

**Comment by o2sh** [comment]
Hi @OlaoluwaM, we use [globbing patterns ](https://docs.rs/globset/latest/globset/) as the format for the `--exclude` pattern, which is the same format used in the [`.gitignore`](https://www.atlassian.com/git/tutorials/saving-changes/gitignore) file, BTW, onefetch will respect all `.gitignore` and `.ignore` files in your repository.

The `--exclude` option affects two aspects:

1. The LOC computation -> "Lines of code" info line and the Languages distribution. You can use this option to exclude both files (ex.: `onefetch -e **/info/*.rs`) and directories (ex.: `onefetch -e src/info`)
2. The churn files -> In that case only files can be excluded via the option.

![image](https://github.com/o2sh/onefetch/assets/13710835/e59a27dd-29d7-4298-b881-87c21b31d88e)
